### PR TITLE
Complete preliminary Draco_Mesh builder, using RTT mesh parser.

### PR DESCRIPTION
**Pull request by RyanWollaeger**
### Background

* In order to test Draco_Mesh on more complicated meshes, some parsing capability is needed.
* There is already mesh parsing functionality in Draco for RTT mesh formats.
* A mesh builder can act as an intermediary between parser and mesh constructor, ensuring all parsers provide the same sort of data to the constructor.

### Purpose of Pull Request

* Create Draco_Mesh_Builder, with reader/parser type as template parameter.
* Demonstrate parsing capability for Draco_Mesh with RTT mesh file format.
* Note the builder is currently assuming a non-decomposed mesh is being read (i.e. global_node_number = proc-local node indices).
* [See Redmine Issue #1182](https://rtt.lanl.gov/redmine/issues/1182)

### Description of changes

+ Add Draco_Mesh_Builder, templated on reader/parser type.
+ Add unit test with 2-cell RTT mesh input file.
+ Add side flag and node coordinate accessors to Draco_Mesh.
+ Fix coordinate generation in Test_Mesh_Interface.
+ Fix header comments in rtt mesh input files.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [ ] Travis CI checks pass
  * [ ] Code coverage does not decrease
  * [ ] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: src/mesh/Draco_Mesh_Builder.hh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+template <typename FRT> class Draco_Mesh_Builder {
+private:
+  // >>> DATA
+
+  //! Pointer to reader
+  std::shared_ptr<FRT> reader;
+
+public:
+  //! Constructor
+  explicit Draco_Mesh_Builder(std::shared_ptr<FRT> reader_);
```
Comment (KineticTheory) [inline src/mesh/Draco_Mesh_Builder.hh]: Please add `DLL_PUBLIC_mesh` for the constructor to fix the error

```
 tstDraco_Mesh_Builder.obj : error LNK2019: unresolved external symbol "public: __thiscall rtt_mesh::Draco_Mesh_Builder<class rtt_RTT_Format_Reader::RTT_Format_Reader>::Draco_Mesh_Builder<class rtt_RTT_Format_Reader::RTT_Format_Reader>(class std::shared_ptr<class rtt_RTT_Format_Reader::RTT_Format_Reader>)" (??0?$Draco_Mesh_Builder@VRTT_Format_Reader@rtt_RTT_Format_Reader@@@rtt_mesh@@QAE@V?$shared_ptr@VRTT_Format_Reader@rtt_RTT_Format_Reader@@@std@@@Z) referenced in function "void __cdecl build_cartesian_mesh_2d(class rtt_c4::ParallelUnitTest &)" (?build_cartesian_mesh_2d@@YAXAAVParallelUnitTest@rtt_c4@@@Z) [C:\projects\build\src\mesh\test\Ut_mesh_tstDraco_Mesh_Builder_exe.vcxproj]
```

File: src/mesh/Draco_Mesh_Builder.hh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  // >>> SERVICES
+
+  std::shared_ptr<Draco_Mesh> build_mesh(rtt_mesh_element::Geometry geometry);
```
Comment (KineticTheory) [inline src/mesh/Draco_Mesh_Builder.hh]: Please add DLL_PUBLIC_mesh here also.

File: src/mesh/Draco_Mesh_Builder.hh (hunk 1 above)
Comment (RyanWollaeger) [reply src/mesh/Draco_Mesh_Builder.hh]: Thank you - this has been added.

File: src/mesh/Draco_Mesh_Builder.hh (hunk 2 above)
Comment (RyanWollaeger) [reply src/mesh/Draco_Mesh_Builder.hh]: This has been added.

**Comment by RyanWollaeger** [comment]
An attempt has been made to fix a double-counting of enumeration declaration caught by CDASH:
https://rtt.lanl.gov/cdash/viewBuildError.php?type=1&buildid=42164.

**Comment by RyanWollaeger** [comment]
It looks like the new unit test timed out on appveyor.

**Comment by KineticTheory** [comment]
@RyanWollaeger That test failure (timeout) is weird. I'll debug it on Friday.

**Comment by RyanWollaeger** [comment]
Thank you to @KineticTheory for the iterator patch.

**Pull request merged**